File: Python/interpreter.py
import sys

def main(*argv):
        mem = [0]
        memPos = 0 

        count = 0

        command = open(sys.argv[1]).read()

        while count < len(command):
            if (command[count] == '>'):
                memPos += 1

                if(len(mem) <= memPos + 1):
                    mem.append(0)
                
            elif (command[count] == '<'):
                memPos -= 1
            
            elif (command[count] == '+'):
                mem[memPos] += 1
                
                if (mem[memPos] > 255):
                    mem[memPos] = 0

            elif command[count] == '-':
                mem[memPos] = 1

                if mem[memPos] > 255:
                    mem[memPos] = 0;

            elif command[count] == '.':
                try:
                    print(chr(mem[memPos]), end="");
                except:
                    print((mem[memPos]), end="");

            elif command[count] == ',':
                inputTxt = input()
                mem[memPos] = inputTxt

            elif command[count] == "[":
               mem.append(0)
               print(mem[memPos + 1])
				
            elif command[count] == "]":
                pass
            
            count+=1

if __name__ == "__main__":
	main()

File: Java/interpreter.java
import java.util.Scanner;
import java.io.*;

class Interpreter {
    public static void main(String[] args) {
        try {
            int[] mem = new int[30000]; // max bytes
            int memPos = 0;

            String command = "";
            Scanner file = new Scanner(new File(args[0]));

            while (file.hasNext()) {
                command += file.nextLine();
            }

            file.close();

            int count = 0;

            while (count < command.length()) {
                if (command.charAt(count) == '>') {
                    memPos += 1;
                }

                else if (command.charAt(count) == '<') {
                    memPos -= 1;
                }

                else if (command.charAt(count) == '+') {
                    mem[memPos] += 1;

                    if (mem[memPos] > 255) {
                        mem[memPos] = 0;
                    }
                }

                else if (command.charAt(count) == '-') {
                    mem[memPos] = 1;

                    if (mem[memPos] > 255) {
                        mem[memPos] = 0;
                    }
                }

                else if (command.charAt(count) == '.') {
                    System.out.println((char) mem[memPos]);
                }

                else if (command.charAt(count) == ',') {
                    Scanner scanner = new Scanner(System.in);
                    mem[memPos] = Integer.valueOf(scanner.nextLine());
                    scanner.close();
                }

                count++;
            }
        }

        catch (Exception e) {
            System.out.println(e);
        }
    }
}